flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerRunner.java

```diff
@@ -103,9 +103,7 @@ public void start() throws Exception {
 	@Override
 	public CompletableFuture<Void> closeAsync() {
 		synchronized (lock) {
```

Comment:
I think this lock can go away completely if you touch this method. Seems that the class can even use a bit more cleanup - lock not needed, field that can be local variable.

Reply:
Yes, I think you are right.